src/lib/y2common_criteria/dialogs/installation.rb

```diff
@@ -0,0 +1,179 @@
+  module Dialogs
+    # Main dialog for the Common Criteria system role, to be presented right after the role
+    # selection
+    class Installation < ::UI::InstallationDialog
+      # Constructor
+      def initialize
+        super
+        textdomain "cc"
+
+        Yast.import "UI"
+        Yast.import "ProductFeatures"
+        Yast.import "Report"
+
+        @passwd_checker = Y2Storage::EncryptPasswordChecker.new
+      end
+
+      # @return [String]
+      def dialog_title
+        _("SUSE Linux Enterprise Common Criteria Evaluated Configuration")
+      end
+
+      # @return [Yast::Term] ui content for dialog
+      def dialog_content
+        entropy = Yast::SCR.Read(
+          Yast::Path.new(".target.string"), "/proc/sys/kernel/random/entropy_avail"
+        ).to_s
+        log.info("entropy #{entropy}")
+
+        HSquash(
+          VSquash(
+            VBox(
+              MinWidth(60, MinHeight(10, RichText(description_text))),
+              VSpacing(2),
+              passwd_widget(:passphrase, _("Encryption Passphrase")),
+              passwd_widget(:repeat_passphrase, _("Verify Passphrase")),
+            )
+          )
+        )
+      end
+
+      # @return [String]
+      def help_text
+        _(
+          "<p>When installing SUSE Linux Enterprise in the Common Criteria evaluated " \
+          "configuration, some restrictions apply to available configuration options.</p>\n" \
+          "<p>Please refer to the deployment guide before proceeding.</p>\n" \
+          "<p>Common Criteria demands all file systems to be encrypted. The passphrase " \
+          "entered here will be used by default for encrypting devices in the partitioning " \
+          "Guided Setup. That includes the initial partitioning proposal automatically " \
+          "calculated by the installer.</p>\n" \
+          "<p>Notice that you will have to enter the correct password each time you boot " \
+          "the system. So make sure to not lose it!</p>"
+        )
+      end
+
+      # Handler for the 'next' event (button)
+      def next_handler
+        return unless valid?
+
+        # put libgcrypt into FIPS mode. libstorage-ng calls the cryptsetup
+        # external command so this takes effect when formatting luks volumes.
+        # Needs to happen in instsys. The target system boots with fips=1 kernel
+        # command line so this setting is not needed.
+        Yast::WFM.Execute(Yast::Path.new(".local.mkdir"), "/etc/gcrypt")
+        Yast::WFM.Write(Yast::Path.new(".local.string"), "/etc/gcrypt/fips_enabled", "1")
+
+        write_passphrase
+
+        super
+      end
+
+      protected
+
+      # @return [Y2Storage::EncryptPasswordChecker]
+      attr_reader :passwd_checker
+
+      # Introductory text to explain how the role works
+      #
+      # @return [String]
+      def description_text
+        _(
+          "<p>Please refer to the deployment guide before proceeding with the " \
```

Comment:
NP: Is the reference to the deployment guide needed here?

I would use a more user-friendly text here, something like:

~~~
Common Criteria demands all file systems to be encrypted.

Enter a passphrase below to be used by default when encrypting devices during system installation.

Read Help for more details.
~~~

And help refers to the deployment guide.

Reply:
See the original screenshot. The reference to the deployment guide was there. Actually, I would say that referencing the deployment guide was the main goal of that screen initially.

Reply:
I tried to keep the essence of the original message while adding the not about the passphrase.